src/main/java/ladder/controller/LadderController.java

```diff
@@ -1,19 +1,82 @@
 package ladder.controller;
 
+import ladder.domain.GameResult;
 import ladder.domain.Ladder;
 import ladder.domain.PlayerNames;
 import ladder.domain.RandomBasedBarGenerator;
 import ladder.view.InputView;
 import ladder.view.OutputView;
 
+import java.util.List;
+
 public class LadderController {
     public void run() {
-        PlayerNames playerNames = InputView.repeat(() -> new PlayerNames(InputView.inputPeopleNames()));
-        int ladderHeight = InputView.repeat(InputView::inputLadderHeight);
-        
-        Ladder ladder = new Ladder(new RandomBasedBarGenerator(), ladderHeight, playerNames.playerSize());
-        
+        PlayerNames playerNames = getCorrectPlayerNames();
+        GameResult gameResult = getGameResult(playerNames);
+    
+        Ladder ladder = getLadder(playerNames);
+    
+        printLadderStructure(playerNames, ladder, gameResult);
+        viewResult(playerNames, ladder, gameResult);
+    }
+    
+    private PlayerNames getCorrectPlayerNames() {
+        return InputView.repeat(() -> new PlayerNames(InputView.inputPeopleNames()));
+    }
+    
+    private GameResult getGameResult(PlayerNames playerNames) {
+        return InputView.repeat(() -> new GameResult(getExecutionResults(), playerNames));
+    }
+    
+    private String getExecutionResults() {
+        return InputView.inputExecutionResults();
+    }
+    
+    private Ladder getLadder(PlayerNames playerNames) {
+        return InputView.repeat(() -> getLadder(playerNames, getCorrectLadderHeight()));
+    }
+    
+    private Integer getCorrectLadderHeight() {
+        return InputView.repeat(InputView::inputLadderHeight);
+    }
+    
+    private Ladder getLadder(PlayerNames playerNames, int ladderHeight) {
+        return new Ladder(new RandomBasedBarGenerator(), ladderHeight, playerNames.playerSize());
+    }
+    
+    private void printLadderStructure(PlayerNames playerNames, Ladder ladder, GameResult gameResult) {
         OutputView.printNames(playerNames);
-        OutputView.printLadder(ladder, playerNames.firstPlayerNameLength());
+        OutputView.printLadder(ladder, playerNames.getFirstPlayerNameLength());
+        OutputView.printExecutionResults(gameResult);
+    }
+    
+    private void viewResult(PlayerNames playerNames, Ladder ladder, GameResult gameResult) {
+        String existedPlayer = getExistedPlayer(playerNames);
+        List<Integer> movedPositions = getMovedPositions(playerNames, ladder);
+    
+        if (existedPlayer.equals("all")) {
```

Comment:
```suggestion
        if ("all".equals(existedPlayer)) {
```
이렇게 하면 null pointer exception을 방지할 수 있답니다.

Reply:
전혀 생각지 못했던 부분이었습니다. null 가능성이 전혀 없는 문자열을 먼저 사용할 생각을 못했네요. 리팩토링 하겠습니다.